ui/plugin-system/src/utils/variables.ts

```diff
@@ -25,23 +26,108 @@ export function replaceVariables(text: string, variableState: VariableStateMap):
     .forEach((v) => {
       const variable = variableState[v];
       if (variable && variable.value !== undefined) {
-        finalText = replaceVariable(finalText, v, variable?.value);
+        finalText = replaceVariable(finalText, v, variable?.value, variablesMap.get(v));
       }
     });
 
   return finalText;
 }
 
-export function replaceVariable(text: string, varName: string, variableValue: VariableValue): string {
+export type InterpolationFormat =
```

Comment:
nit: This could be an enum, and to get the type we could use
```
type LogLevelStrings = keyof typeof LogLevel;
```
(example from https://www.typescriptlang.org/docs/handbook/enums.html#enums-at-compile-time)

Then we could drop the stringToFormat, which duplicates all format options.